include/envoy/network/transport_socket.h

```diff
@@ -137,6 +137,11 @@ class TransportSocket {
    * @return the const SSL connection data if this is an SSL connection, or nullptr if it is not.
    */
   virtual const Ssl::Connection* ssl() const PURE;
+
+  /**
+   * @return the TransportSocketCallbacks object passed in through setTransportSocketCallbacks().
+   */
+  virtual TransportSocketCallbacks* callbacks() PURE;
```

Comment:
I would remove this as I think where you need get `callbacks` is in the scope you set it. No strong opinion on it though. If this is needed internally, ping me on hangout.

Reply:
done